File: ruby-numerals/numerals_util.rb

class NumeralsUtil
	def self.numeral_right_part num, cut_pos
		num.to_s[cut_pos..-1].to_i
	end

	def self.numeral_left_part num, cut_pos
		num.to_s[0..cut_pos-1].to_i
	end

	def self.numeral_fraction num, cut_pos_begin, cut_pos_end
		num.to_s[cut_pos_begin..cut_pos_end-1].to_i
	end

	def self.add_end_space_if_required partial_transl, space_req
		add_end_something_if_required(partial_transl, space_req, ' ')
	end

	def self.add_end_minus_if_required partial_transl, space_req
		add_end_something_if_required(partial_transl, space_req, '-')
	end

	def self.add_end_something_if_required partial_transl, space_req, something
		if (space_req) 
			partial_transl = partial_transl + something
		end
		partial_transl
	end

	def self.add_prefix_if_not_null partial_transl, prefix
		if (partial_transl.to_s.length > 0) 
			partial_transl = partial_transl + prefix
		end
		partial_transl
	end

	def self.add_begin_space_if_not_null partial_transl
		add_begin_something_if_not_null(partial_transl, ' ')
	end

	def self.add_begin_AND_if_not_null partial_transl
		add_begin_something_if_not_null(partial_transl, 'and ')
	end

	def self.add_begin_something_if_not_null partial_transl, something
		if (partial_transl.to_s.length > 0) 
			partial_transl = something + partial_transl
		end
		partial_transl
	end

	def self.add_AND_before_last_not_null_part part1, part2, part3
		if (part3.to_s.length > 0)
			part3 = ' and' + part3
		elsif (part2.to_s.length > 0)
			part2 = ' and' + part2
		end
		part1 + part2 + part3
	end
	def self.clear_number dirty_number
		dirty_number.to_s.gsub(/[^0-9.]/, "").to_i
	end
end

File: js-numerals/numeralsUtil.js

function numeralRightPart(num, cutPos) {
	return parseInt(num.toString().substring(cutPos, num.toString().length));
}
function numeralLeftPart(num, cutPos) {
	return parseInt(num.toString().substring(0, cutPos));
}
function numeralFraction(num, cutPosBegin, cutPosEnd) {
	return parseInt(num.toString().substring(cutPosBegin, cutPosEnd));
}
function addEndSpaceIfRequired(partialTransl, spaceReq) {
	return addEndSomethingIfRequired(partialTransl, spaceReq, ' ');
}
function addEndMinusIfRequired(partialTransl, spaceReq) {
	return addEndSomethingIfRequired(partialTransl, spaceReq, '-');
}
function addEndSomethingIfRequired(partialTransl, spaceReq, something) {
	if (spaceReq) {
		partialTransl = partialTransl + something;
	}
	return partialTransl;
}
function addPrefixIfNotNull(partialTransl, prefix) {
	if (isEmpity(partialTransl)) {
		partialTransl = partialTransl + prefix
	}
	return partialTransl;
}
function addBeginSpaceIfNotNull(partialTransl) {
	return addBeginSomethingIfNotNull(partialTransl, ' ');
}
function addBeginANDIfNotNull(partialTransl) {
	return addBeginSomethingIfNotNull(partialTransl, 'and ');
}
function addBeginSomethingIfNotNull(partialTransl, something) {
	if (isEmpity(partialTransl)) { 
		partialTransl = something + partialTransl;
	}
	return partialTransl;
}
function addANDBeforeLastNotNullPart(part1, part2, part3) {
	if (isEmpity(part3)) {
		part3 = ' and' + part3;
	} else if (isEmpity(part2)) {
		part2 = ' and' + part2;
	}
	return part1 + part2 + part3;
}
function isEmpity(s) {
	return s != null && s.length > 0;
}
function clearNumber (dirtyNum) {
	// var partialValue = Math.floor(dirtyNum)
	var partialValue = dirtyNum.toString().replace(/[^\d.]/g, '');
	if (partialValue == '') { 
		partialValue = 0;
	}
	return parseInt(partialValue);
}
